gqcp/include/Mathematical/Optimization/OptimizationEnvironment.hpp

```diff
@@ -0,0 +1,60 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include <deque>
+
+
+namespace GQCP {
+
+
+/**
+ *  A standard environment that stores the iterative variables of an iterative algorithm
+ * 
+ *  @tparam _Iterate            the type of the iterative variables
+ */
+template <typename _Iterate>
+class OptimizationEnvironment {
+public:
+    using Iterate = _Iterate;
+    using Scalar = typename Iterate::Scalar;
+
+
+public:
+    std::deque<_Iterate> variables;
```

Comment:
Is there anymore context you could give this variable? Given it starts with a guess it is only used as an input value for an equation that is solved iteratively? Do I understand this correctly?

Reply:
I've added some more context in the documentation. Thanks for pointing it out.

As usual, the point is that the `environment` grows with every iteration that is performed. In this case, `variables` represents a general `x` when we are solving `f(x) = 0` for example.